File: seed/csharp-model/objects-with-imports/src/SeedObjectsWithImportsClient/File/Directory/Directory.cs
using System.Text.Json.Serialization
using SeedObjectsWithImportsClient
using SeedObjectsWithImportsClient.File

namespace SeedObjectsWithImportsClient.File

public class Directory
{
    [JsonPropertyName("name")]
    public string Name { get; init; }
    [JsonPropertyName("files")]
    public List<File>? Files { get; init; }
    [JsonPropertyName("directories")]
    public List<Directory>? Directories { get; init; }
}

File: seed/rust-model/examples/src/directory.rs
use serde::{Deserialize, Serialize};
use crate::file::File;

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct Directory {
    pub name: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub files: Option<Vec<File>>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub directories: Option<Vec<Directory>>,
}